source/features/highest-rated-comment.tsx

```diff
@@ -63,6 +93,27 @@ function init(): false | void {
 	}
 }
 
+function getCommentsWithReactions(): Set<HTMLElement> {
+	// Map reaction [aria-label*=...] to .js-item [aria-label*=...], then join and pass to select.all
+	const reactions = select.all(positiveReactions.map(reaction => `${COMMENT_SELECTOR} ${reaction}`).join(','));
+	// Find closest comment to each reaction, cast to HTMLElement, then filter out null values
+	const comments = reactions.map(reaction => reaction.closest(COMMENT_SELECTOR))
+		.map(comment => comment as HTMLElement).filter(comment => comment);
```

Comment:
It's impossible that there are null values since the initial selector has `COMMENT_SELECTOR`

To fix the types, this just needs to be

```ts
function getCommentsWithReactions(): Set<HTMLElement> {
	const comments = getPositiveReactions().map(reaction => reaction.closest<HTMLElement>(COMMENT_SELECTOR)!);
	return new Set(comments);
}
```

And you can include `COMMENT_SELECTOR` in the original `positiveReactions` array.

Also probably you can avoid the `HTMLElement` altogether:



```ts
function getCommentsWithReactions(): Set<Element> {
	const comments = getPositiveReactions().map(reaction => reaction.closest(COMMENT_SELECTOR)!);
	return new Set(comments);
}
```

Reply:
I got a syntax error with the first way, I looked at the source and I think it only allows for types that extend certain bases to be passed as type parameters. 
I went with the second way instead.

Reply:
The code still doesn't match my suggestion

Reply:
The suggested code did not compile without some default parameters and I didn't know that document could be a default parameter because in C# default parameters must be compile time constant.